lib/segment/src/vector_storage/raw_scorer.rs

```diff
@@ -160,12 +160,14 @@ fn new_scorer_with_metric<'a, TMetric: Metric + 'a, TVectorStorage: DenseVectorS
 ) -> Box<dyn RawScorer + 'a> {
     let vec_deleted = vector_storage.deleted_vector_bitslice();
     match query {
-        QueryVector::Nearest(vector) => raw_scorer_from_query_scorer(
-            MetricQueryScorer::<TMetric, _>::new(vector, vector_storage),
-            point_deleted,
-            vec_deleted,
-            is_stopped,
-        ),
+        QueryVector::Nearest(vector) => match vector {
```

Comment:
Here I propose to use `into()` because we will change `raw_scorer_from_query_scorer` signature and may forget to remove this match